elf2cfetbl.c

```diff
@@ -1895,13 +1895,13 @@ void PrintSectionHeader32(union Elf_Shdr *SectionHeader)
 
 void PrintSectionHeader64(union Elf_Shdr *SectionHeader)
 {
-    if (Verbose) printf("   sh_addr       = 0x%lx\n", SectionHeader->Shdr64.sh_addr);
-    if (Verbose) printf("   sh_offset     = 0x%08lx\n", SectionHeader->Shdr64.sh_offset);
-    if (Verbose) printf("   sh_size       = 0x%08lx\n", SectionHeader->Shdr64.sh_size);
+    if (Verbose) printf("   sh_addr       = 0x%lx\n", (long unsigned int)SectionHeader->Shdr64.sh_addr);
+    if (Verbose) printf("   sh_offset     = 0x%08lx\n", (long unsigned int)SectionHeader->Shdr64.sh_offset);
+    if (Verbose) printf("   sh_size       = 0x%08lx\n", (long unsigned int)SectionHeader->Shdr64.sh_size);
     if (Verbose) printf("   sh_link       = 0x%08x\n", SectionHeader->Shdr64.sh_link);
     if (Verbose) printf("   sh_info       = 0x%08x\n", SectionHeader->Shdr64.sh_info);
```

Comment:
For instance wouldn't the "sh_link" and "sh_info" need the same cast?  (these are also words of different lengths depending on arch/binary format).

Reply:
Clarified pull request, urgent fix is for vxworks6.9, MCP750.  As you mention, likely requires more casts for other systems but not a certification priority.

Reply:
How is this related to VxWorks exactly?  The tool runs on the host machine, not on VxWorks.  Are the mismatches specific to running it on a 32-bit host machine vs 64-bit host machine?

Reply:
True enough, this is a 32 bit host.  I only fixed the issues that showed up for the vxworks build on the 32 bit host.  I should have better described the test system.

Reply:
OK, that makes more sense.  But I'm still not an advocate of just minimally squelching the visible warnings on a specific 32-bit host  (IIRC the machine with the VxWorks 6.9 tools is a 32-bit RHEL 6.x host).  Other compilers/libraries might flag other printfs.  Unless we actually add all the needed casts, warnings are likely reappear again as soon as someone builds on another system.

Reply:
It's a step in the right direction, and removes a cert blocker.  Making it work across compilers/libraries is a very valid open source concern, but not my priority.